services/src/main/scala/cromwell/services/keyvalue/impl/KeyValueServiceActor.scala

```diff
@@ -31,14 +29,27 @@ case class KeyValueServiceActor(serviceConfig: Config, globalConfig: Config) ext
     }
   }
 
-  private def doPut(put: KvPut): Future[KvResponse] = Future.successful {
-    store += put.pair.key -> put.pair.value
-    KvPutSuccess(put)
+  private def doPut(put: KvPut): Future[KvResponse] = {
+    put.pair.value match {
+      case Some(backendVal) => updateBackendKeyValuePair(put.pair.key.workflowId,
+        put.pair.key.jobKey,
+        put.pair.key.key,
+        put.pair.value.get).map(_ => KvPutSuccess(put))
+      case None => Future(KvFailure(put, new Throwable(s"Failed to find the value associated to key: ${put.pair.key.key}. This key cannot be added to the BackendKVStore.")))
```

Comment:
Never create a `new Throwable`. (side rant: anyone know why that class isn't abstract?) It's dangerous for handling code to try and catch them, as `java.lang.Error`, the sibling of `Exception`, shouldn't be normally handled.

Use `new RuntimeException` or even `new Exception` if you must save on typing characters. The other instances in cromwell and wdl4s that are doing this should all be fixed, someday.